ts/features/bonus/cgn/screens/merchants/CgnMerchantDetailScreen.tsx

```diff
@@ -215,14 +166,34 @@ const CgnMerchantDetailScreen = () => {
                 value={merchantDetail.value.description}
               />
               <VSpacer size={24} />
-              {renderMerchantAddressesList(
-                merchantDetail.value.addresses,
-                merchantDetail.value.allNationalAddresses
-              )}
+              {showAddresses || showGotToWebsite ? (
+                <ListItemHeader
+                  label={I18n.t("bonus.cgn.merchantDetail.title.contactInfo")}
+                />
+              ) : null}
+              {showGotToWebsite ? (
+                <ListItemAction
+                  variant="primary"
+                  icon="website"
+                  label={I18n.t("bonus.cgn.merchantDetail.cta.website")}
+                  onPress={() =>
+                    handlePressMerchantWebsite(merchantDetail.value.websiteUrl)
+                  }
+                />
+              ) : null}
+              {showGotToWebsite && showAddresses ? <Divider /> : null}
+              {merchantDetail.value.addresses?.map((address, index) => (
+                <CgnAddressListItem
+                  item={address}
```

Comment:
Using array indices as keys in React can lead to issues with component state and performance, especially when the list can change. Instead, you should use a unique identifier for each item.

Reply:
unfortunately address is as embedded value, no unique identifiers available from backend as of now

Reply:
I saw we had installed the uuid library. In that case I suggest you to use it to generate unique keys for each item instead of relying on array indices

Reply:
that would be using random keys, index is just as good as a random value

Reply:
Oh correct. Could the address serve as a unique identifier for each merchant, or do you think that could be problematic due to multiple businesses sharing the same location?

Reply:
I do not think a partner would ever put 2 identical addresses. But if they to, using full_address as key would result in a bug

Reply:
@Hantex9  what are your thoughts? 
Currently, I don’t have a global scope on the domain

Reply:
I don't have a strong opinion on this, but since there can only be up to 3 addresses, performance edge cases are not a concern here. Especially since the backend doesn't provide an ID, I think using the index or a UUID would be equally fine in this case. However, I would avoid using the address name as a key, because—even though it's unlikely—there's still a very small chance that a merchant could enter duplicate addresses (even by mistake).

What do you think?

Reply:
Both using the index or a UUID would work, but I would lean toward UUID for flexibility and to avoid potential issues with reordering or modifying the address list

Reply:
At the moment, there is no action involving modifying, reordering, or expanding the data (there will always be a maximum of 3 addresses), so using the index is not incorrect. However, to be future-proof, in case at some distant point we decide to introduce a more dynamic way of handling these addresses, like reordering or similar changes, we could consider using UUIDs. What do you think?

Reply:
We are on the same page